movie-app/src/shared-components/switchers-buttons/SwitchersButtons.tsx

```diff
@@ -10,9 +10,14 @@ const SwitchersButtons = (props: any) => {
         i === index ? (item.active = true) : (item.active = false);
         return item;
       });
+
+      if (props.click && props.activeBtnValue !== arr[index].buttonValue) {
+        props.click(arr[index].buttonValue);
+      }
+
       updateSearchButtons(arr);
     },
-    [buttons]
+    [props.activeBtnValue],
```

Comment:
`click` and `buttons` are missing at dependencies

Reply:
Need to discuss